src/main/java/org/akip/service/NotifyUserAssignedTasksService.java

```diff
@@ -0,0 +1,57 @@
+package org.akip.service;
+
+import org.akip.domain.TaskInstance;
+import org.akip.domain.enumeration.StatusTaskInstance;
+import org.akip.repository.TaskInstanceRepository;
+import org.akip.resolver.AkipUserDTO;
+import org.akip.resolver.UserResolver;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+public class NotifyUserAssignedTasksService {
+
+    private final AkipMailService mailService;
+
+    private final TaskInstanceRepository taskInstanceRepository;
+
+    private final UserResolver userResolver;
+
+    public NotifyUserAssignedTasksService(AkipMailService mailService, TaskInstanceRepository taskInstanceRepository, UserResolver userResolver) {
+        this.mailService = mailService;
+        this.taskInstanceRepository = taskInstanceRepository;
+        this.userResolver = userResolver;
+    }
+
+    public Map<String, String> notifyUserAssignedTasks(){
+        Map<String, String> summary = new HashMap<>();
+        Map<String, Object> variables = new HashMap<>();
```

Comment:
Move to line 50. In general, you should postpone the declaration of your variables as much as possible.